Add more settings for Indices Request Cache (#7160)
opensearch-project/documentation-website: _search-plugins/caching/tiered-cache.md

To use the OpenSearch-provided tiered spillover cache implementation, set the cache store name to `tiered_spillover`, as shown in the following example:

```yaml

> indices.request.cache.store.name: tiered_spillover

Author: This is a bug

The following table lists the cache store settings for the `opensearch_onheap` store.

> | Setting                                          | Data type  | Default                | Description                                                      |

Author: okay, I reverted in later commits, thanks !

| `indices.request.cache.ehcache_disk.expire_after_access` | Time unit | `MAX_VALUE` (disabled)                   | Specifies a TTL for the cached results. Optional.                                                                                 |
| `indices.request.cache.ehcache_disk.alias`               | String    | `ehcacheDiskCache#INDICES_REQUEST_CACHE` | Specifies an alias for the disk cache. Optional.                                                                                  |
| `indices.request.cache.ehcache_disk.segments`            | Integer   | `16`                                     | Defines the number of segments into which the disk cache is separated. Used for concurrency. Optional.                            |
| `indices.request.cache.ehcache_disk.concurrency`         | Integer   | `1`                                      | Defines the number of distinct write queues created for the disk store, where a group of segments shares a write queue. Optional. |

### Additional settings for the `tiered_spillover` store

The following table lists additional settings for the `tiered_spillover` store setting.

> | Setting                                                                          | Data type | Default | Description                                                                                                                                                                                                        |

Reviewer: ```suggestion
|  Setting  | Data type | Default | Description |
```

> |:---------------------------------------------------------------------------------|:----------|:--------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

Reviewer: ```suggestion
| :--- |:--- | :--- | :--- |
```

> | `indices.request.cache.tiered_spillover.disk.store.policies.took_time.threshold` | Time unit | `10ms`  | A policy used to determine whether to cache a query into a disk cache based on its took time. This is a dynamic setting. Optional.                                                                                 |

Reviewer: ```suggestion
| `indices.request.cache.tiered_spillover.disk.store.policies.took_time.threshold` | Time unit | `10ms`  | A policy used to determine whether to cache a query into a disk cache based on its took time. This is a dynamic setting. Optional.  |
```

> | `indices.request.cache.tiered_spillover.disk.store.enabled`                      | Boolean   | `True`  | Enables or disables the disk cache dynamically within a tiered spillover cache. Note: After disabling a disk cache, entries are not removed automatically and requires the cache to be manually cleared. Optional. |

Author: @hdhalter these were recommended by IntelliJ IDE

Reviewer 2: OK, TFLMK. it looks exactly the same for me.

> | `indices.request.cache.tiered_spillover.disk.store.enabled`                      | Boolean   | `True`  | Enables or disables the disk cache dynamically within a tiered spillover cache. Note: After disabling a disk cache, entries are not removed automatically and requires the cache to be manually cleared. Optional. |

Reviewer: ```suggestion
| `indices.request.cache.tiered_spillover.disk.store.enabled`  | Boolean   | `True`  | Enables or disables the disk cache dynamically within a tiered spillover cache. Note: After disabling a disk cache, entries are not removed automatically and requires the cache to be manually cleared. Optional. |
```

### Configuring cleanup

> The following table lists the settings related to deletion of stale entries from the cache.

Reviewer: ```suggestion
The following table lists the settings related to the deletion of stale entries from the cache.
```

> | Setting                                              | Data type | Default | Description                                                                                                 |

Reviewer: ```suggestion
| Setting | Data type | Default | Description |
```

> |:-----------------------------------------------------|:----------|:--------|:------------------------------------------------------------------------------------------------------------|

Reviewer: ```suggestion
| :--- | :--- | :--- | :--- |
```

> | `indices.requests.cache.cleanup.staleness_threshold` | String    | `0%`    | Defines the percentage of stale keys in the cache post which all stale cache entries are deleted. Optional. |

Reviewer: ```suggestion
| `indices.requests.cache.cleanup.staleness_threshold` | String    | `0%`    | Defines the percentage of stale keys in the cache post. All stale cache entries are deleted. Optional. |
```

Author: I dont think `post. All` makes sense. 
What i meant to say is
Defines the percentage of stale keys in the cache **after** which all stale cache entries are deleted

> | `indices.requests.cache.cleanup.interval`            | Time unit | `1m`    | Defines the frequency at which the request cache's stale entries are deleted. Optional.                     |

Author: These are the changes I'm introducing as new settings

> | `indices.requests.cache.cleanup.interval`            | Time unit | `1m`    | Defines the frequency at which the request cache's stale entries are deleted. Optional.                     |

Reviewer: ```suggestion
| `indices.requests.cache.cleanup.interval` | Time unit |  `1m`  | Defines the frequency at which the request cache's stale entries are deleted. Optional.  |
```

:--- | :--- | :--- | :---
`indices.request.cache.tiered_spillover.disk.store.policies.took_time.threshold` | Time unit | `10ms` | A policy used to determine whether to cache a query into a disk cache based on its took time. This is a dynamic setting. Optional.
`indices.request.cache.tiered_spillover.disk.store.enabled` | Boolean | `True` | Enables or disables the disk cache dynamically within a tiered spillover cache. Note: After disabling a disk cache, entries are not removed automatically and requires the cache to be manually cleared. Optional.

> ### Configuring cleanup

Reviewer: ```suggestion
### Delete stale entries settings
```

The following table lists the settings related to the deletion of stale entries from the cache.

Setting | Data type | Default | Description
:--- | :--- |:--------| :---

> `indices.requests.cache.cleanup.staleness_threshold` | String | `0%`    | Defines the percentage of stale keys in the cache post which all stale cache entries are deleted. Optional.

Reviewer: ```suggestion
`indices.requests.cache.cleanup.staleness_threshold` | String | `0%`    | Defines the percentage of stale keys in the cache post. After identification, all stale cache entries are deleted. Optional.
```
